tests/integration/targets/aci_endpoint_ip_tag/tasks/main.yml

```diff
@@ -0,0 +1,418 @@
+
+- name: Assertion check for remove IP Tag Object with ip and vrf - check_mode
+  ansible.builtin.assert:
+    that:
+      - cm_rm_with_ip_and_vrf is changed
+      - cm_rm_with_ip_and_vrf.current.0.fvEpIpTag.attributes.dn == "uni/tn-ansible_test/eptags/epiptag-[2.2.2.2]-ansible_test_vrf"
+      - cm_rm_with_ip_and_vrf.previous.0.fvEpIpTag.attributes.dn == "uni/tn-ansible_test/eptags/epiptag-[2.2.2.2]-ansible_test_vrf"
+
+- name: Remove IP Tag Object with ip and vrf - normal mode
+  cisco.aci.aci_endpoint_ip_tag:
+    <<: *aci_info
+    tenant: ansible_test
+    endpoint_ip_address: "2.2.2.2"
+    vrf: ansible_test_vrf
+    state: absent
+  register: nm_rm_with_ip_and_vrf
+
+- name: Assertion check for remove IP Tag Object with ip and vrf - normal mode
+  ansible.builtin.assert:
+    that:
+      - nm_rm_with_ip_and_vrf is changed
+      - nm_rm_with_ip_and_vrf.previous.0.fvEpIpTag.attributes.annotation == "orchestrator:ansible"
+      - nm_rm_with_ip_and_vrf.previous.0.fvEpIpTag.attributes.ctxName == "ansible_test_vrf"
+      - nm_rm_with_ip_and_vrf.previous.0.fvEpIpTag.attributes.dn == "uni/tn-ansible_test/eptags/epiptag-[2.2.2.2]-ansible_test_vrf"
+      - nm_rm_with_ip_and_vrf.previous.0.fvEpIpTag.attributes.ip == "2.2.2.2"
+      - nm_rm_with_ip_and_vrf.previous.0.fvEpIpTag.attributes.name == ""
+      - nm_rm_with_ip_and_vrf.previous.0.fvEpIpTag.attributes.nameAlias == ""
+
+- name: Remove IP Tag Object with ip and vrf - normal mode again
+  cisco.aci.aci_endpoint_ip_tag:
+    <<: *aci_info
+    tenant: ansible_test
+    endpoint_ip_address: "2.2.2.2"
+    vrf: ansible_test_vrf
+    state: absent
+  register: nm_rm_with_ip_and_vrf_again
+
+- name: Assertion check for remove IP Tag Object with ip and vrf - normal mode again
+  ansible.builtin.assert:
+    that:
+      - nm_rm_with_ip_and_vrf_again is not changed
+      - nm_rm_with_ip_and_vrf_again.current == []
+      - nm_rm_with_ip_and_vrf_again.previous == []
+
+- name: Query all existing Endpoint IP Tags after the test
+  cisco.aci.aci_endpoint_ip_tag:
+    <<: *aci_info
+    state: query
+  register: query_all_ip_tags_1
+
+- name: Assertion check for query all existing Endpoint IP Tags after the test
+  ansible.builtin.assert:
+    that:
+      - query_all_ip_tags_1 is not changed
+      - query_all_ip_tags_1.current != []
+      - query_all_ip_tags_1.current | length >= 2
+
+- name: Clear existing Endpoint IP Tags from ansible_test tenant after the test
+  cisco.aci.aci_endpoint_ip_tag:
+    <<: *aci_info
+    tenant: ansible_test
+    endpoint_ip_address: "{{ item.fvEpIpTag.attributes.ip }}"
+    vrf: "{{ item.fvEpIpTag.attributes.ctxName }}"
+    state: absent
+  when: "{{ item.fvEpIpTag.attributes.dn is match('uni/tn-ansible_test/.+?')  }}"
+  loop: "{{ query_all_ip_tags_1.current | list }}"
+
+- name: Ensure ansible_test_vrf VRF not exists
+  cisco.aci.aci_vrf:
+    <<: *aci_info
+    tenant: ansible_test
+    vrf: ansible_test_vrf
+    state: absent
```

Comment:
these could be skipped if you delete the tenant and it is only for cleanup, since deleting the parent will delete all children